docs/ops/sort/TopK_11.md

```diff
@@ -0,0 +1,118 @@
+# TopK  {#openvino_docs_ops_sort_TopK_11}
+
+**Versioned name**: *TopK-11*
+
+**Category**: *Sorting and maximization*
+
+**Short description**: *TopK* computes indices and values of the *k* maximum/minimum values for each slice along a specified axis.
+
+**Attributes**
+
+* *axis*
+
+  * **Description**: Specifies the axis along which the values are retrieved.
+  * **Range of values**: An integer. Negative values means counting dimension from the back.
+  * **Type**: `int`
+  * **Required**: *yes*
+
+* *mode*
+
+  * **Description**: Specifies whether *TopK* selects the largest or the smallest elements from each slice.
+  * **Range of values**: "min", "max"
+  * **Type**: `string`
+  * **Required**: *yes*
+
+* *sort*
+
+  * **Description**: Specifies the order of corresponding elements of the output tensor.
+  * **Range of values**: `value`, `index`, `none`
+  * **Type**: `string`
+  * **Required**: *yes*
+
+* *stable*
+
+  * **Description**: Specifies whether the equivalent elements should maintain their relative order from the input tensor. Takes effect only if the `sort` attribute is set to `value`.
+  * **Range of values**: `true` of `false`
+  * **Type**: `boolean`
+  * **Default value**: `false`
+  * **Required**: *no*
+
+* *index_element_type*
+
+  * **Description**: the type of output tensor with indices
+  * **Range of values**: "i64" or "i32"
+  * **Type**: string
+  * **Default value**: "i32"
+  * **Required**: *no*
+
+
+**Inputs**:
+
+*   **1**: tensor with arbitrary rank and type *T*. **Required.**
+
+*   **2**: The value of *K* -- a scalar of any integer type that specifies how many elements from the input tensor should be selected. **Required.**
+
+**Outputs**:
+
+*   **1**: Output tensor of type *T* with at most *k* values from the input tensor along specified dimension *axis*. The shape of the tensor is `[input1.shape[0], ..., input1.shape[axis-1], 1..k, input1.shape[axis+1], ..., input1.shape[input1.rank - 1]]`.
```

Comment:
can you please elaborate how we pad the output if `k` greater `input1.shape[axis]`?

Reply:
TF, PT and ONNXRT throw an exception and cancel the inference in such case. I think we should do the same.

Reply:
The CPU plugin throws an exception and doesn't allow an inference with K value larger than the dimension at axis. The GPU plugin does not throw and it returns a tensor with exactly K values.

Reply:
I think we need to say about the undefined behavior in this case explicitly. This is important to cover this corner case in the spec explicitly. It will simplify understanding of this specification. Moreover, the different plugins behave differently for this corner case now so this statement will be aligned with it.

Reply:
I've added this information to the second input description. It now defines a range of allowed values of K.